test/index.test.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-env mocha */
+
+'use strict'
+
+const assert = require('assert')
+const { promisify } = require('util')
+
+const createGrapes = require('bfx-svc-test-helper/grapes')
+const createFxGrenache = require('bfx-svc-test-helper/fauxgrenache')
+
+const Fac = require('..')
+
+const stubs = {
+  'rest:util:net': {
+    getIpInfo (space, ip, cb) {
+      const res = [ip, { country: 'US', region: 'CA' }]
+      return cb(null, res)
+    }
+  }
+}
+
+const grapes = createGrapes()
+const caller = {
+  on: () => {},
+  ctx: { root: __dirname }
+}
+
+const fac = new Fac(caller, {})
+fac.tick = () => {}
+fac.conf = { grape: 'http://127.0.0.1:30001', transport: 'http' }
+
+let fxg
+describe('RPC integration', () => {
+  before(async function () {
+    this.timeout(20000)
+
+    grapes.onAnnounce()
+
+    await grapes.start()
+    fxg = createFxGrenache(stubs, grapes)
+    await fxg.start()
+
+    const start = promisify(fac.start).bind(fac)
+    await start()
+  })
+
+  after(async function () {
+    this.timeout(5000)
+
+    await grapes.stop()
+    await fxg.stop()
+    fac.stop()
+  })
+
+  describe('_doReq', () => {
```

Comment:
lets have one file per feature that is tested, else we end up with 500 LOC testfiles where you scroll all day

Reply:
i'm testing a single function here, which is `_doReq`. Do you want me to split callback / promise tests to different files?